utils/numbers/validations.ts

```diff
@@ -0,0 +1,4 @@
+export function canBecomeInteger(value: unknown) {
+  const number = Number(value);
+  return !isNaN(number) && Number.isInteger(number);
```

Comment:
Actually "falsey" values like 0, "", and undefined return 0 when passed to `Number()`

```suggestion
  return (!value && value !== 0) && !isNaN(number) && Number.isInteger(number);
```

Reply:
Sorry my bad! Didn't read your description. I am in favour of guarding against these cases within this function.